webapp/src/auth.ts

```diff
@@ -505,36 +584,44 @@ function authApiHandler(p: string) {
     const field = data.stripProtocol(p);
     const state = getState();
     switch (field) {
-        case FIELD_USER: return state.user;
+        case FIELD_PROFILE: return state.profile;
         case FIELD_LOGGED_IN: return loggedInSync();
     }
     return null;
 }
 
+/**
+ * Clear all auth state.
+ * Direct access to state$ allowed.
+ */
 function clearState() {
-    state_ = {};
-    data.invalidate(USER);
-    data.invalidate(LOGGED_IN);
+    state$ = {};
+    pxt.storage.removeLocal(AUTH_USER_STATE);
+    data.invalidate("auth:*");
+    //data.invalidate("user-prefs:*"); // should we clear this?
 }
```

Comment:
Should this be moved up next to the other state$ accessors ?

Reply:
moved!